### PR TITLE
support application/octet-stream in proxy_bridge

### PR DESCRIPTION
**Pull request by aaminev**
needed for this https://github.com/zenefits/yourPeople3/pull/6468

### CODE REVIEW

File: lib/proxy_bridge.js [hunk 1]
```diff
@@ -24,11 +24,14 @@ var Bridge = function (channelUuid) {
...
-    var contentType = req.headers['content-type'];
+    var contentType = req.headers['content-type'] || '';
     // Buffer the body if needed
-    return ((req.method == 'POST' || req.method == 'PUT') &&
-      (contentType && (contentType.indexOf('form-urlencoded') > -1 ||
-      contentType.indexOf('application/json') > -1)));
+    return (req.method == 'POST' || req.method == 'PUT') &&
+      (
+        contentType.indexOf('form-urlencoded') > -1 ||
+        contentType.indexOf('application/json') > -1) ||
```
Comment (asiunov) [inline lib/proxy_bridge.js]: one redundant bracket in this line

**Comment by aaminev** [comment]
@zen-mde please review this PR

**Comment by zen-mde** [comment]
Can we please do a change that's a little narrower in scope? We are wanting to work toward decoupling the frontend from the backend, and a general proxy means even tighter coupling, because people will be less likely to mock out external network services.

**Comment by zen-esundstr** [comment]
@aaminev Can we maybe have it only work for the octet-stream for particular urls?  Maybe put that in a json somewhere.

File: lib/proxy_bridge.js [hunk 2]
```diff
@@ -21,16 +21,24 @@ var Bridge = function (channelUuid) {
   this.inFlight = {};
 };
 
+
+//If there are other urls for which octet-stream is a valid content type 
+//   add regular expressions for their urls here:
+var octetStreamUrlWhitelist = [ /custom_api\/documents\/in_memory_s3\/([\w\-_]+)/([\w\-_\/]+)$/ ];
```
Comment (zen-mde) [inline lib/proxy_bridge.js]: Variables should be declared at the top of their respective scopes. Also, would prefer this to be named in CONSTANT_CASE. :D

File: lib/proxy_bridge.js [hunk 3]
```diff
@@ -21,14 +21,25 @@ var Bridge = function (channelUuid) {
...
 Bridge.prototype = new (function () {
 
   var _expectBody = function (req) {
-    var contentType = req.headers['content-type'];
+    var contentType = req.headers['content-type'] || '';
     // Buffer the body if needed
-    return ((req.method == 'POST' || req.method == 'PUT') &&
-      (contentType && (contentType.indexOf('form-urlencoded') > -1 ||
-      contentType.indexOf('application/json') > -1)));
+    var validOctetStream = !octetStreamUrlWhitelist.every(function(re) {
```
Comment (zen-mde) [inline lib/proxy_bridge.js]: Instead of not-not every, you can just use `some`.

Comment (asiunov) [inline lib/proxy_bridge.js]: Why is this not a function? Maybe not so important, but in most cases this check will not be processed.

**Pull request merged**